shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -588,6 +593,13 @@ - (UIView*)keyboardAnimationView {
   return _keyboardAnimationView.get();
 }
 
+- (UIScreen*)mainScreen {
+  if (@available(iOS 13.0, *)) {
+    return self.viewIfLoaded.window.windowScene.screen;
```

Comment:
should the function be called `mainScreenIfLoadedView`?

Reply:
Hmm, I'm hesitant to change it to that since in older version of the OS, it does not rely on the view. Is that a common naming convention we do?

Reply:
if you change it to `self.view` will it crash? (and when?)

im wondering when do we need to access the keyboard logic if the view is not loaded.

Reply:
So using `self.view` was crashing due to the leaking test (https://github.com/flutter/engine/pull/37719/commits/39313a7739605ddadb8654f6ace90b111a9cd97a). After fixing the test, though, it would not crash. However, I changed to `viewIfLoaded` to prevent any future crashes.

Reply:
this may cause bugs in the future that is very tricky to debug. can you either crash or log here?

Reply:
`view` already crashes if it's not loaded, that's why `viewIfLoaded` exists. https://developer.apple.com/documentation/uikit/uiviewcontroller/1621360-viewifloaded?language=objc

See https://github.com/flutter/flutter/issues/112545 and https://github.com/flutter/engine/pull/36580 for what that looks like.

Reply:
Is `FML_DLOG(INFO)` appropriate to use? Seems to me `FML_DLOG` is used for debug logs. Or would `FML_LOG(INFO)` be better?

Reply:
`FML_LOG(WARNING)` is probably better, since when this function is called before `viewDidLoad`, something unexpected must happen.

Reply:
updated